streams/src/main/java/org/apache/kafka/streams/kstream/Materialized.java

```diff
@@ -97,21 +128,21 @@ protected Materialized(final Materialized<K, V, S> materialized) {
         this.cachingEnabled = materialized.cachingEnabled;
         this.topicConfig = materialized.topicConfig;
         this.retention = materialized.retention;
-        this.storeType = materialized.storeType;
+        this.storeSuppliers = materialized.storeSuppliers;
     }
 
     /**
-     * Materialize a {@link StateStore} with the given {@link StoreType}.
+     * Materialize a {@link StateStore} with the given {@link DslStoreSuppliers}.
      *
-     * @param storeType  the type of the state store
-     * @param <K>       key type of the store
-     * @param <V>       value type of the store
-     * @param <S>       type of the {@link StateStore}
+     * @param storeSuppliers  the type of the state store
+     * @param <K>             key type of the store
+     * @param <V>             value type of the store
+     * @param <S>             type of the {@link StateStore}
      * @return a new {@link Materialized} instance with the given storeName
      */
-    public static <K, V, S extends StateStore> Materialized<K, V, S> as(final StoreType storeType) {
-        Objects.requireNonNull(storeType, "store type can't be null");
-        return new Materialized<>(storeType);
+    public static <K, V, S extends StateStore> Materialized<K, V, S> as(final DslStoreSuppliers storeSuppliers) {
```

Comment:
Just to clarify (for myself as well as anyone else reviewing this), we are able to simply swap in the new `DSLStoreSuppliers` parameter -- instead of deprecating this and adding a new API -- because source-code compatibility is maintained due to the new `StoreType` inheritance.

Reply:
yup!